src/libraries/System.Text.Json/src/System.Text.Json.csproj

```diff
@@ -9,7 +9,7 @@
     <IncludeInternalObsoleteAttribute>true</IncludeInternalObsoleteAttribute>
     <IsPackable>true</IsPackable>
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
-    <ServicingVersion>1</ServicingVersion>
+    <ServicingVersion>2</ServicingVersion>
```

Comment:
@eiriktsarpalis @tarekgh I bumped the version to 2.

Since we built a version 1 last month (https://github.com/dotnet/runtime/pull/94882), the `GeneratePackageOnBuild` property was already set to true, so we don't need to modify it.

I submitted a separate PR to reset all the OOB packages that were built last month, and I skipped S.T.J. on purpose since we will build it again this month. https://github.com/dotnet/runtime/pull/96704